src/libraries/ProtocolFeeLibrary.sol

```diff
@@ -20,11 +22,19 @@ library ProtocolFeeLibrary {
         if (self != 0) {
             uint16 fee0 = getZeroForOneFee(self);
             uint16 fee1 = getOneForZeroFee(self);
-            // The fee is represented in bips so it cannot be GREATER than the MAX_PROTOCOL_FEE.
+            // The fee is represented in pips and it cannot be greater than the MAX_PROTOCOL_FEE.
             if ((fee0 > MAX_PROTOCOL_FEE) || (fee1 > MAX_PROTOCOL_FEE)) {
                 return false;
             }
         }
         return true;
     }
+
+    // Effective fee can never exceed 100% (1e6 pips)
+    function calculateEffectiveFee(uint24 self, uint24 swapFee) internal pure returns (uint24) {
```

Comment:
maybe we can comment the original formula?
// the lp fee is applied after the protocol fee, and the total fee is capped at 100%
// swapFee = protocolFee * lpFee(1_000_000 - protocolFee) / 1_000_000